apphub/automl/rand_augment/rand_augment_tf.py

```diff
@@ -0,0 +1,310 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import random
+import tempfile
+
+import numpy as np
+import tensorflow as tf
+from PIL import Image, ImageEnhance, ImageOps, ImageTransform
+from tensorflow.python.keras import layers
+
+import fastestimator as fe
+from fastestimator.dataset.data.cifar10 import load_data
+from fastestimator.op.numpyop import NumpyOp
+from fastestimator.op.numpyop.meta import OneOf
+from fastestimator.op.numpyop.univariate import Normalize
+from fastestimator.op.tensorop.loss import CrossEntropy
+from fastestimator.op.tensorop.model import ModelOp, UpdateOp
+from fastestimator.trace.metric import Accuracy
+
+
+def residual(x, num_channel):
+    x = layers.Conv2D(num_channel, 3, padding='same')(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    x = layers.Conv2D(num_channel, 3, padding='same')(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    return x
+
+
+def my_model():
+    # prep layers
+    inp = layers.Input(shape=(32, 32, 3))
+    x = layers.Conv2D(64, 3, padding='same')(inp)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    # layer1
+    x = layers.Conv2D(128, 3, padding='same')(x)
+    x = layers.MaxPool2D()(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    x = layers.Add()([x, residual(x, 128)])
+    # layer2
+    x = layers.Conv2D(256, 3, padding='same')(x)
+    x = layers.MaxPool2D()(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    # layer3
+    x = layers.Conv2D(512, 3, padding='same')(x)
+    x = layers.MaxPool2D()(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    x = layers.Add()([x, residual(x, 512)])
+    # layers4
+    x = layers.GlobalMaxPool2D()(x)
+    x = layers.Flatten()(x)
+    x = layers.Dense(10)(x)
+    x = layers.Activation('softmax', dtype='float32')(x)
+    model = tf.keras.Model(inputs=inp, outputs=x)
+
+    return model
+
+
+class Rotate(NumpyOp):
```

Comment:
if the augmentation is the same between tf and torch, do we want to make a rand_augment_ops.py file for them? or keep everything inlined?

Reply:
having two files is better, so that it is consistent with other apphubs.